src/hapi/pipelines/database/food_security.py

```diff
@@ -23,138 +23,189 @@ def __init__(
-                    for key in population_in_phases:
-                        population_total = sum(
-                            filter(None, population_totals[key[1:]])
+                        if any(
+                            x in admintwo_name.lower()
+                            for x in food_sec_config["adm_ignore_patterns"]
+                        ):
+                            add_message(
+                                warnings,
+                                dataset_name,
+                                f"Admin 2: ignoring {full_adm2name}",
+                            )
+                            continue
+                        admintwo, exact = self._admintwo.get_pcode(
+                            countryiso3, admintwo_name, parent=adminone
                         )
-                        population_in_phase = sum(population_in_phases[key])
-                        food_security_row = DBFoodSecurity(
-                            resource_hdx_id=resource_id,
-                            admin2_ref=admin2_ref,
-                            ipc_phase=key[0],
-                            ipc_type=key[1],
-                            reference_period_start=key[2],
-                            reference_period_end=key[3],
-                            population_in_phase=population_in_phase,
-                            population_fraction_in_phase=(
-                                population_in_phase / population_total
-                                if population_in_phase > 0
-                                else 0.0
-                            ),
+                        if not admintwo:
+                            add_message(
+                                warnings,
+                                dataset_name,
+                                f"Admin 2: could not match {full_adm2name}!",
+                            )
+                            continue
+                        if not exact:
+                            name = self._admintwo.pcode_to_name[admintwo]
+                            if admintwo_name in food_sec_config["adm2_errors"]:
+                                add_message(
+                                    errors,
+                                    dataset_name,
+                                    f"Admin 2: ignoring erroneous {full_adm2name} match to {name} {(admintwo)}!",
+                                )
+                                continue
+                            add_message(
+                                warnings,
+                                dataset_name,
+                                f"Admin 2: matching {full_adm2name} to {name} {(admintwo)}",
+                            )
+                        admin2_ref = self.get_admin2_ref(
+                            subnational_level, admintwo, dataset_name, errors
                         )
-                        self._session.add(food_security_row)
-        self._session.commit()
-
+                    else:
```

Comment:
I think this else is redundant

Reply:
I added it because PyCharm complains admin2_ref might be referenced before assignment without it.

Reply:
After my refactoring it is in any case gone